doc/proxy-metrics.md

```diff
@@ -0,0 +1,69 @@
++++
+title = "Proxy Metrics"
+docpage = true
+[menu.docs]
+  parent = "proxy-metrics"
++++
+
+The Conduit proxy exposes metrics that describe the traffic flowing through the
+proxy.  These metrics are exposed in Prometheus format on the proxy control port
+(4190 by default) at the `/metrics` path.  The following metrics will be
+exposed:
+
+### `request_total`
+
+A counter of the number of requests the proxy has received.
+
+### `request_duration_ms`
+
+A histogram of the duration of a request.  This is measured from when the
+request headers are received to when the request stream has completed.
+
+### `response_total`
+
+A counter of the number of responses the proxy has received.
+
+### `response_duration_ms`
+
+A histogram of the duration of a response.  This is measured from when the
+response headers are received to when the response stream has completed.
+
+### `response_latency_ms`
+
+A histogram of the total latency of a response.  This is measured from when the
+request headers are received to when the response stream has completed.
+
+## Labels
+
+Each of these metrics has the following labels:
+
+* `direction`: `inbound` if the request originated from outside of the pod,
+               `outbound` if the request originated from inside of the pod.
+* `deployment`: The deployment that the pod belongs to (if applicable).
+* `job`: The job that the pod belongs to (if applicable).
+* `replica_set`: The replica set that the pod belongs to (if applicable).
+* `daemon_set`: The daemon set that the pod belongs to (if applicable).
+* `replication_controller`: The replication controller that the pod belongs to 
+                            (if applicable).
+* `authority`: The value of the `:authority` (HTTP/2) or `Host` (HTTP/1.1)
+               header of the request.
+* `dst_deployment`: The deployment to which this request is being sent.  Only
+                    applicable if `direction=outbound`.
```

Comment:
realizing this was [discussed](https://github.com/runconduit/conduit/issues/425#issuecomment-370946692) in a previous issue, but seeing all these `applicable if direction=outbound` in this doc makes me think we should put direction back into the metric name. with separate inbound and outbound metrics, we can still do math on them as needed, but it will avoid some complexity of evaluating labels based on the presence of other labels.

Reply:
I could go either way on this.

Reply:
@siggy are there situations where the direction label causes a problem? can you share an example?

Reply:
this design does not preclude functionality, though. it does make some query patterns ambiguous. for example, if we want to count outbound responses to `foo`, which query is appropriate:

```
response_total{dst_deployment="foo"}
response_total{direction="outbound", dst_deployment="foo"}
```
with `outbound_response_total`, there's only one way:
```
outbound_response_total{dst_deployment="foo"}
```